pkg/utils/common.go

```diff
@@ -109,3 +117,44 @@ func SanitizeRepoUrl(url string) string {
 
 	return url
 }
+
+func CreateAppSecretName(targetName string, repoURL string) string {
+	return fmt.Sprintf("wego-%s-%s", targetName, UrlToRepoName(repoURL))
+}
+
+func StartK8sTestEnvironment() (client.Client, error) {
+	testEnv := &envtest.Environment{
+		CRDDirectoryPaths: []string{
+			"../../manifests/crds",
+			"../../tools/testcrds",
+		},
+	}
+
+	var err error
+	cfg, err := testEnv.Start()
+	if err != nil {
+		return nil, fmt.Errorf("could not start testEnv: %w", err)
+	}
+
+	scheme := kube.CreateScheme()
+
+	k8sManager, err := ctrl.NewManager(cfg, ctrl.Options{
+		ClientDisableCacheFor: []client.Object{
+			&wego.Application{},
+			&corev1.Namespace{},
+			&corev1.Secret{},
+		},
+		Scheme: scheme,
+	})
+	if err != nil {
+		return nil, fmt.Errorf("could not create controller manager: %w", err)
+	}
+
+	go func() {
+		if err := k8sManager.Start(ctrl.SetupSignalHandler()); err != nil {
+			fmt.Printf("error starting k8s manager: %s", err.Error())
```

Comment:
Since this function is about starting a k8s test environment, we should fail if we can't start the manager instead of just printing a message.

Reply:
Fixed. Calling the `Expect` func in the test helper